autoscheduler/frontend/src/timeUtil.ts

```diff
@@ -8,7 +8,13 @@ export const LAST_HOUR = 21;
  * @param m minutes
  * @param use24Hour if true, applies 24-hour format instead of the default 12-hour format
  */
-export function formatTime(h: number, m: number, use24Hour = false): string {
-  return `${use24Hour ? h : (h - 1) % 12 + 1}:${
+export function formatTime(h: number, m: number, use24Hour = false, padZeroes = false): string {
+  const ret = `${use24Hour ? h : (h - 1) % 12 + 1}:${
     new Intl.NumberFormat('en-US', { minimumIntegerDigits: 2 }).format(m)}`;
+
+  if (padZeroes && ret.length < 4) {
+    return `0${ret}`;
+  }
+
+  return ret;
```

Comment:
This function is a bit messy right now, if you reuse the NumberFormat for hours it's more straightforward:
```suggestion
  const formattedHours = use24Hour ? h : (h - 1) % 12 + 1;
  const padZero = new Intl.NumberFormat('en-US', { minimumIntegerDigits: 2 });
  return `${padZeroes ? padZero.format(formattedHours) : formattedHours}:${padZero.format(m)}`;
```